src/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Security
+{
+    internal class SslAuthenticationOptions
+    {
+        internal SslAuthenticationOptions(SslClientAuthenticationOptions sslClientAuthenticationOptions)
+        {
+            AllowRenegotiation = sslClientAuthenticationOptions.AllowRenegotiation;
+            ApplicationProtocols = sslClientAuthenticationOptions.ApplicationProtocols;
+            CheckCertificateRevocation = sslClientAuthenticationOptions.CheckCertificateRevocation;
+            ClientCertificates = sslClientAuthenticationOptions.ClientCertificates;
+            EnabledSslProtocols = sslClientAuthenticationOptions.EnabledSslProtocols;
+            EncryptionPolicy = sslClientAuthenticationOptions.EncryptionPolicy;
+            LocalCertificateSelectionCallback = sslClientAuthenticationOptions.LocalCertificateSelectionCallback;
+            RemoteCertificateValidationCallback = sslClientAuthenticationOptions.RemoteCertificateValidationCallback;
+            CertValidationDelegate = sslClientAuthenticationOptions._certValidationDelegate;
+            CertSelectionDelegate = sslClientAuthenticationOptions._certSelectionDelegate;
+            TargetHost = sslClientAuthenticationOptions.TargetHost;
+            IsServer = false;
+            CheckCertName = true;
+            RemoteCertRequired = true;
+        }
+
+        internal SslAuthenticationOptions(SslServerAuthenticationOptions sslServerAuthenticationOptions)
+        {
+            TargetHost = string.Empty;
+            AllowRenegotiation = sslServerAuthenticationOptions.AllowRenegotiation;
+            RemoteCertRequired = sslServerAuthenticationOptions.ClientCertificateRequired;
+            ApplicationProtocols = sslServerAuthenticationOptions.ApplicationProtocols;
+            RemoteCertificateValidationCallback = sslServerAuthenticationOptions.RemoteCertificateValidationCallback;
+            CertValidationDelegate = sslServerAuthenticationOptions._certValidationDelegate;
+            ServerCertificate = sslServerAuthenticationOptions.ServerCertificate;
+            EnabledSslProtocols = sslServerAuthenticationOptions.EnabledSslProtocols;
+            CheckCertificateRevocation = sslServerAuthenticationOptions.CheckCertificateRevocation;
+            EncryptionPolicy = sslServerAuthenticationOptions.EncryptionPolicy;
+            CheckCertName = false;
+            IsServer = true;
+        }
```

Comment:
It would be nice if the client and server copy ctors copied the fields in the same order.